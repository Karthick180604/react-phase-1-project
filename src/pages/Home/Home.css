:root{
    --primary: #FAFAFA;
    --secondary:#2C5E4C;
    --tertiary:#E7C961;
}
*{
    margin: 0px;
    padding: 0px;
}

.home-container {
  background: linear-gradient(to bottom, var(--tertiary), var(--primary));
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.home-content-section {
  height: 100%;
  width: 40%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 40px;
}
.home-welcome-text {
  font-size: 28px;
  line-height: 1.6;
  font-weight: 400;
  color: #333;
  text-align: left;
  max-width: 600px;
}

.brand-name {
  color:var(--secondary);
  font-weight: 600;
}

.home-image-section {
  height: 100%;
  width: 40%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}


.home-image-container {
  height: 300px;
  width: 300px;
  border-radius: 50%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: 0 8px 20px rgba(0, 0, 0, 0.15);
}

.home-image-container img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

@media(max-width:768px){
  .home-container{
    display: flex;
    flex-direction: column-reverse;
  }
  .home-welcome-text{
    font-size: 18px;
  }
  .home-image-container{
    height: 150px;
    width: 150px;
  }
  .home-content-section{
    width: 80%;
  }
}