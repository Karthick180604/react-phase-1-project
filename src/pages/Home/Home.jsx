import React, { Component } from "react"
import welcomeImage from "../../assets/shutter-speed-BQ9usyzHx_w-unsplash.jpg"
import "./Home.css"

class Home extends Component{
    render(){
        return(
            <div className="home-container">
                <div className="home-content-section">
                    <p className="home-welcome-text">Welcome to <span className="brand-name">Shopora</span>! where style meets tech, elegance meets performance. 
                    <br/>
                    Discover the best in fashion, gadgets, jewelry, and more all in one place, just for you.</p>
                </div>
                <div className="home-image-section">
                    <div className="home-image-container">
                        <img src={welcomeImage}></img>
                    </div>
                </div>
            </div>
        )
    }
}

export default Home