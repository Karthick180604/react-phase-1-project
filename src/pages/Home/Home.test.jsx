import { render, screen } from "@testing-library/react";
import Home from "./Home";
import "@testing-library/jest-dom";

test("renders welcome message", () => {
  render(<Home />);
  
  const paragraphs = screen.getAllByText((_, element) =>
    element?.textContent?.includes(
      "Welcome to Shopora! where style meets tech, elegance meets performance."
    )
  );
  const secondLine = screen.getByText(/Discover the best in fashion, gadgets, jewelry/i);
  
  expect(paragraphs[0]).toBeInTheDocument();
  expect(secondLine).toBeInTheDocument();
});

test("renders welcome image", () => {
  render(<Home />);
  
  const img = screen.getByRole("img");
  expect(img).toBeInTheDocument();
  expect(img).toHaveAttribute("src", expect.stringContaining("shutter-speed"));
});

test("home snapshot",()=>{
  const home=render(<Home />)
  expect(home).toMatchSnapshot();
})
