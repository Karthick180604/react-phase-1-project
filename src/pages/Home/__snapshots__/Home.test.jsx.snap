// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-container"
      >
        <div
          class="home-content-section"
        >
          <p
            class="home-welcome-text"
          >
            Welcome to 
            <span
              class="brand-name"
            >
              Shopora
            </span>
            ! where style meets tech, elegance meets performance.
            <br />
            Discover the best in fashion, gadgets, jewelry, and more all in one place, just for you.
          </p>
        </div>
        <div
          class="home-image-section"
        >
          <div
            class="home-image-container"
          >
            <img
              src="/src/assets/shutter-speed-BQ9usyzHx_w-unsplash.jpg"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-container"
    >
      <div
        class="home-content-section"
      >
        <p
          class="home-welcome-text"
        >
          Welcome to 
          <span
            class="brand-name"
          >
            Shopora
          </span>
          ! where style meets tech, elegance meets performance.
          <br />
          Discover the best in fashion, gadgets, jewelry, and more all in one place, just for you.
        </p>
      </div>
      <div
        class="home-image-section"
      >
        <div
          class="home-image-container"
        >
          <img
            src="/src/assets/shutter-speed-BQ9usyzHx_w-unsplash.jpg"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
